analyses/snv-callers/compare_snv_callers_plots.Rmd

```diff
@@ -131,141 +145,155 @@ source(file.path("util", "full_join_callers.R"))
 
 # Bring out the data.frame
 all_caller_df <- all_caller %>%
-  as.data.frame
+  as.data.frame() %>%
+  # We'll use this to keep track of the original rows/mutations
+  tibble::rowid_to_column("index")
 
 # Take a peek at what this looks like
 head(all_caller_df)
 ```
 
+## Get consensus numbers
+
+Make a data.frame with only the VAFs and turn into detected or not. 
+
 ```{r}
-# Set up the data as a data.frame
-vaf_df <- all_caller_df %>% 
-  dplyr::select("VAF_lancet", "VAF_strelka", "VAF_vardict", "VAF_mutect",
-                "Variant_Classification") %>%
-  dplyr::mutate(VAF_consensus = VAF_strelka)     
- 
-# Determine which mutations are part of the consensus
-consensus_index <- vaf_df %>%
-  dplyr::select(-c(VAF_vardict, VAF_consensus, Variant_Classification)) %>%
-  rowSums(is.na(.)) > 0
-
-# Make non consensus VAF into an NA
-vaf_df$VAF_consensus[consensus_index] <- NA
-
-# Make this long form for plotting and 
-long_vaf_df <- vaf_df %>% 
-  dplyr::mutate(index = factor(1:nrow(.))) %>% 
-  tidyr::gather(key = "caller", value = "vaf", -index, -Variant_Classification) %>% 
-  dplyr::mutate(caller = gsub("VAF_", "", caller)) %>% 
-  dplyr::filter(!is.na(vaf))
+detect_df <- all_caller_df %>% 
+ # Bring over VAF columns and the index
+ dplyr::select(dplyr::starts_with("VAF_"), index) %>% 
+ # Turn into detected or not
+ dplyr::mutate_at(dplyr::vars(dplyr::starts_with("VAF_")), function(x) !is.na(x))
 ```
 
-Get median vaf for each mutation with each caller combination. 
+Count up how many callers call each mutation but ignore VarDict. 
 
 ```{r}
-# Take out the consensus VAF for this combination
-long_caller_df <- long_vaf_df %>% 
-  dplyr::filter(caller != "consensus")
-
-# Determine the combination of callers for each mutation
-callers_per_mutation <- tapply(
-    long_caller_df$caller,
-    long_caller_df$index,
-    paste0,
-    collapse = "-"
-  ) %>% 
-  as.data.frame() %>%
-  tibble::rownames_to_column("index")
-
-# Determine Median VAF for each mutation 
-vaf_med <- tapply(
-    long_caller_df$vaf,
-    long_caller_df$index,
-    median
-  ) %>% 
-  as.data.frame() %>%
-  tibble::rownames_to_column("index")
-
-# Join the median VAF and the callers that call that mutation into one data.frame
-callers_per_mutation <- callers_per_mutation %>%
-  dplyr::inner_join(vaf_med, by = "index") %>%
-  dplyr::left_join(long_caller_df %>% 
-                      dplyr::select(Variant_Classification, index), 
-                   by = "index") %>%
-  # Make column names more sensible
-  dplyr::rename(caller_combo = "..x", median_vaf = "..y") 
+consensus_count <- detect_df %>%
+  # Ignore VarDict and index
+  dplyr::select(-VAF_vardict, -index) %>%
+  # Get the number of callers that call each mutation
+  dplyr::transmute(num = rowSums(.)) %>% 
```

Comment:
This step is quite a memory hog and will not run on even a 192 GB size AWS container for PBTA.